java/code/src/com/suse/manager/metrics/SystemsCollector.java

```diff
@@ -0,0 +1,50 @@
+package com.suse.manager.metrics;
+
+import com.redhat.rhn.domain.server.Server;
+import com.redhat.rhn.domain.server.ServerFactory;
+import com.redhat.rhn.manager.system.SystemManager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.prometheus.client.Collector;
+
+public class SystemsCollector extends Collector {
+
+    private static final String PRODUCT_NAME = "uyuni";
+
+    private List<Server> servers;
+
+    @Override
+    public List<MetricFamilySamples> collect() {
+        List<MetricFamilySamples> out = new ArrayList<>();
+        servers = ServerFactory.lookupByIds(SystemManager.listSystemIds());
+
+        out.add(CustomCollectorUtils.gaugeFor("managed_systems", "Number of managed systems",
+                getNumberOfSystems(), PRODUCT_NAME));
+        out.add(CustomCollectorUtils.gaugeFor("physical_systems", "Number of managed physical systems",
+                getNumberOfPhysicalSystems(), PRODUCT_NAME));
+        out.add(CustomCollectorUtils.gaugeFor("inactive_systems", "Number of inactive managed systems",
+                getNumberOfInactiveSystems(), PRODUCT_NAME));
+        out.add(CustomCollectorUtils.gaugeFor("outdated_systems", "Number of systems with outdated packages",
+                getNumberOfOutdatedSystems(), PRODUCT_NAME));
+
+        return out;
+    }
+
+    private long getNumberOfSystems() {
+        return servers.size();
```

Comment:
You return all systems known. But we have caterories where we can argue if this is really a 'managed' system.
E.g. Foreign or Bootstrap. Also what about containerized proxy? Is this 'managed'?

Reply:
Do you see it as the naming problem? In the [prototype exporter](https://github.com/bjin01/exporters/blob/69e41745443e0bf151fb13076d5117001f320d7a/exporter.go#L252-L277) we have `all_systems = physical_systems + virtual_systems`. Is it a valid assumption or should we further differentiate?

Reply:
Thanks, updated.